contrib/systemd/asterisk.service

```diff
@@ -16,19 +17,53 @@ Group=asterisk
 ExecStart=/usr/sbin/asterisk -mqf -C /etc/asterisk/asterisk.conf
 ExecReload=/usr/sbin/asterisk -rx 'core reload'
 #if /var/run is a tmpfs, this will create /var/run/asterisk on start
-#RuntimeDirectory=asterisk
+RuntimeDirectory=asterisk
 
+# Standard priority for Asterisk process
 #Nice=0
+# Prioritize Asterisk process before other processes on the system
+#Nice=-20
+
 #UMask=0002
 LimitCORE=infinity
+
+# More files means more connections allowed
 #LimitNOFILE=
+# This is reasonable for appx. 100 PJSIP extensions (circa 2024)
+#LimitNOFILE=65536
+# You can view the current limit via Asterisk CLI "core show settings"
+
 Restart=always
 RestartSec=4
 
 # Prevent duplication of logs with color codes to /var/log/messages
 StandardOutput=null
 
+#
+# SANDBOXING
+#
+
+# PrivateTmp means Asterisk will use private /tmp and /var/tmp
 PrivateTmp=true
 
+# ProtectHome is recommended in the systemd.exec manual page for
+# "...long-running services (in particular network-facing ones)..."
+# This restricts any Asterisk access to /home, /root and /run/user
+ProtectHome=true
+
+# Mount the entire file system read-only, except for /dev, /proc, /sys
+# and any exceptions (see ReadWritePaths below for some exception ideas)
+ProtectSystem=strict
+
+# Minimalist operation - PID and ctl files, AstDB, Record(), and Logs
+ReadWritePaths=/var/run /var/lib/asterisk /var/log/asterisk
```

Comment:
When RuntimeDirectory is specified and `/var/run` is a tmpfs, systemd will create the directory before applying the protection.  If `/var/run` isn't a tmpfs then someone will have to have created `/var/run/asterisk` manually.  Either way,  there should be no need for RW on `/var/run` itself so it can be further scoped down to `/var/run/asterisk`.

I also think you need to include `/var/spool/asterisk` in the default config.  Voicemail and call recording also require it and those are fairly common.

Have you checked that these all still work if runuser and rungroup are set to a non-privileged user in asterisk.conf?  I don't know why they wouldn't but I'm just playing safe.

Reply:
Excellent points, thank you. Will add limit to `/var/run/asterisk`. Tested that it works with asterisk/asterisk user/group on recent local Debian 12 VM. (Not sure if testsuite addresses systemd issues.)